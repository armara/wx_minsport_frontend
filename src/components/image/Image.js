import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import 'components/image/Image.scss'

const Image = ({ src, alt, className }) => (
  <img src={src} alt={alt} className={classnames('base-image', className)} />
)

Image.defaultProps = {
  alt: '',
  className: '',
}

Image.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string,
  className: PropTypes.string,
}

export default Image
