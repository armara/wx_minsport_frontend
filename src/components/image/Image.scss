.base-image {
  object-fit: contain;
}