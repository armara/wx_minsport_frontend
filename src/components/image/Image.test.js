import React from 'react'
import Image from 'components/image/Image'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

configure({ adapter: new Adapter() })

describe('<Image />', () => {
  it('renders correctly default props', () => {
    const tree = renderer.create(<Image src="hello" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
