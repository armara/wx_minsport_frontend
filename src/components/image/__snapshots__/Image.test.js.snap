// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> renders correctly default props 1`] = `
<img
  alt=""
  className="base-image"
  src="hello"
/>
`;
